tools/mkdocs_macros_bids/macros.py

```diff
@@ -347,3 +347,20 @@ def define_common_principles(src_path=None):
     schema_obj = schema.load_schema()
     string = render.define_common_principles(schema_obj, src_path=src_path)
     return string
+
+
+def define_allowed_top_directories():
+
+    schema_obj = schema.load_schema()
+
+    text = ""
+
+    for key in schema_obj["objects"]["associated_data"]:
+
+        required = schema_obj["rules"]["associated_data"][key]["required"]
+        status = "REQUIRED" if required == True else "OPTIONAL"
+        description = schema_obj["objects"]["associated_data"][key]["description"]
+
+        text += f"- `{key}`: [{status}] {description}"
```

Comment:
@tsalo do you think the core of this macro should actually go into the render module of the schema tools?

Reply:
I think that would be preferable

Reply:
okidoki

will give this a go... but tomorrow... now is 💤 time...

Reply:
Done